engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/DelayTransformer.scala

```diff
@@ -0,0 +1,83 @@
+package pl.touk.nussknacker.engine.flink.util.transformer
+
+
+import java.time.Duration
+
+import org.apache.flink.api.common.state.{ValueState, ValueStateDescriptor}
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.streaming.api.functions.KeyedProcessFunction
+import org.apache.flink.streaming.api.scala._
+import org.apache.flink.streaming.runtime.operators.windowing.TimestampedValue
+import org.apache.flink.util.Collector
+import pl.touk.nussknacker.engine.api
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.flink.api.compat.ExplicitUidInOperatorsSupport
+import pl.touk.nussknacker.engine.flink.api.process.{FlinkCustomNodeContext, FlinkCustomStreamTransformation}
+import pl.touk.nussknacker.engine.flink.util.keyed.StringKeyOnlyMapper
+
+object DelayTransformer extends CustomStreamTransformer with ExplicitUidInOperatorsSupport {
+
+  @MethodToInvoke(returnType = classOf[Void])
+  def invoke(@ParamName("key") key: LazyParameter[CharSequence],
```

Comment:
I started to wonder - do we really need the key to be passed by user explicitly? 
We need it internally - but isn't it enough to generate random number or sth like that? 
We used similar element at one of the clients and they struggled to understand what the key should be for this element....

Reply:
I've made key optional, with default value - empty string. Also used MapState, so reading state won't hurt us even for one key